.travis.yml

```diff
@@ -19,3 +19,16 @@ env:
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
 
 script : .travis.sh
+
+jobs:
+    include:
+        - stage: Upload Package
+          # We need to include the exclusion of D2 tags because this "if"
+          # replaces the global one
+          if: tag IS present
+          # Which package to deploy
+          env: DMD=dmd DIST=xenial
+          # before_install: and install: are inherited and we don't need to
+          # override them.
+           sudo: required
+           script : beaver docker build . && beaver docker run make pkg && beaver bintray upload -N build/last/pkg/*.deb
```

Comment:
Ideally `beaver docker build` should go to the `install:` part. Some other parts should probably be moved from the general config to the stage too, to avoid messing with the more pure-travis approach of the rest of the `.travis.yml` file, for example the `before_install:` and `git:` parts (not sure if all can be moved there as I never tried it myself). You are missing the `-d` argument to the upload command. Also to build the package you should use `beaver dlang make`, not `run make`.
```yml
jobs:
    include:
        - stage: Build & Upload Package
          if: tag IS present
          # Which package to deploy
          env:
              - DMD=2.077.*
              - DIST=xenial
              - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
          language: generic
          sudo: required
          services:
              - docker
          git:
              submodules: false
          before_install: git submodule update --init
          install: beaver docker build .
          script:
              - beaver docker run make pkg
              - beaver bintray upload -N -d dlang-community/apt build/last/pkg/*.deb
```